src/solvers/strings/string_constraint_generator.h

```diff
@@ -74,392 +74,248 @@ class string_constraint_generatort final
-  const typet &type,
-  const bool strict_formatting,
-  const array_string_exprt &str,
-  const std::size_t max_string_length,
-  const exprt &radix,
-  const unsigned long radix_ul,
-  array_poolt &array_pool);
-string_constraintst add_axioms_for_correct_number_format(
-  const array_string_exprt &str,
-  const exprt &radix_as_char,
-  const unsigned long radix_ul,
-  const std::size_t max_size,
-  const bool strict_formatting,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_parse_int(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool,
-  const namespacet &ns);
-std::pair<exprt, string_constraintst> add_axioms_for_compare_to(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
+  std::pair<exprt, string_constraintst>
+  add_axioms_for_last_index_of(const function_application_exprt &f);
 
-exprt is_digit_with_radix(
-  const exprt &chr,
-  const bool strict_formatting,
-  const exprt &radix_as_char,
-  const unsigned long radix_ul);
+  /// \todo The specifications of these functions is only partial.
+  /// We currently only specify that the string for NaN is "NaN", for infinity
+  /// and minus infinity the string are "Infinity" and "-Infinity respectively
+  /// otherwise the string contains only characters in [0123456789.] and '-' at
+  /// the start for negative number
+  std::pair<exprt, string_constraintst>
+  add_axioms_for_string_of_float(const function_application_exprt &f);
+  std::pair<exprt, string_constraintst>
+  add_axioms_for_string_of_float(const array_string_exprt &res, const exprt &f);
+  std::pair<exprt, string_constraintst> add_axioms_for_fractional_part(
+    const array_string_exprt &res,
+    const exprt &int_expr,
+    size_t max_size);
+  std::pair<exprt, string_constraintst>
+  add_axioms_from_float_scientific_notation(
+    const array_string_exprt &res,
+    const exprt &f);
+  std::pair<exprt, string_constraintst>
+  add_axioms_from_float_scientific_notation(
+    const function_application_exprt &f);
 
-exprt get_numeric_value_from_character(
-  const exprt &chr,
-  const typet &char_type,
-  const typet &type,
-  const bool strict_formatting,
-  unsigned long radix_ul);
+  /// Add axioms corresponding to the String.valueOf(D) java function
+  /// \todo The specifications is only partial.
+  std::pair<exprt, string_constraintst>
+  add_axioms_from_double(const function_application_exprt &f);
 
-size_t max_printed_string_length(const typet &type, unsigned long ul_radix);
+  std::pair<exprt, string_constraintst>
+  add_axioms_for_replace(const function_application_exprt &f);
+  std::pair<exprt, string_constraintst>
+  add_axioms_for_set_length(const function_application_exprt &f);
+
+  /// \todo The specification may not be correct for the case where the
+  /// string is shorter than end. An actual java program should throw an
+  /// exception in that case.
```

Comment:
:bulb: this looks outdated compared to the `add_axioms_for_substring(const function_application_exprt)` (but not related to this PR)